browser/ui/tabs/split_view_browser_data.h

```diff
@@ -0,0 +1,66 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_TABS_SPLIT_VIEW_BROWSER_DATA_H_
+#define BRAVE_BROWSER_UI_TABS_SPLIT_VIEW_BROWSER_DATA_H_
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "base/types/pass_key.h"
+#include "chrome/browser/ui/browser_user_data.h"
+#include "chrome/browser/ui/tabs/tab_model.h"
+
+class SplitViewTabStripModelAdapter;
+
+class SplitViewBrowserData : public BrowserUserData<SplitViewBrowserData> {
+ public:
+  using TilePassKey = base::PassKey<SplitViewTabStripModelAdapter>;
+
+  using Tile = std::pair<tabs::TabHandle, tabs::TabHandle>;
+
+  ~SplitViewBrowserData() override;
+
+  // When calling this, make sure that tile.first is the left tab and
+  // tile.second is the right tab. This order will be persistent across the all
+  // tab strip model operations.
+  void TileTabs(const Tile& tile);
+
+  void BreakTile(const tabs::TabHandle& tab);
+
+  bool IsTabTiled(const tabs::TabHandle& tab) const;
+
+  const Tile& GetTile(const tabs::TabHandle& tab) const;
```

Comment:
I think because `std::optional` is implicitly castable to a boolean you can just write:

```cpp
if (auto tile = GetTile()) {
    // do something
}
```

Reply:
Wow, I thought it'd work only for pointer type. Thanks